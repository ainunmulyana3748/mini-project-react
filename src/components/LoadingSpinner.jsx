import React from "react";

const LoadingSpinner = () => {
  return (
    <span className="relative w-12 h-12 rounded-full border-4 border-white border-t-transparent animate-spin inline-block box-border">
      <span className="absolute inset-0 m-auto w-10 h-10 border-4 border-b-red-600 border-l-red-600 border-t-transparent border-r-transparent rounded-full animate-[spin_0.5s_linear_infinite]"></span>
      <span className="absolute inset-0 m-auto w-8 h-8 border-4 border-white border-t-transparent border-r-transparent rounded-full animate-[spin_1.5s_linear_infinite]"></span>
    </span>
  );
};

export default LoadingSpinner;
