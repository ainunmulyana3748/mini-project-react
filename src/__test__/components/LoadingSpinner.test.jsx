import { render } from "@testing-library/react";
import LoadingSpinner from "../../components/LoadingSpinner";

describe("LoadingSpinner Component", () => {
  test("render spinner", () => {
    const { container } = render(<LoadingSpinner />);
    // Cek ada elemen dengan class animate-spin
    expect(container.querySelector(".animate-spin")).toBeInTheDocument();
  });

  test("snapshot spinner", () => {
    const { container } = render(<LoadingSpinner />);
    expect(container).toMatchSnapshot();
  });
});
