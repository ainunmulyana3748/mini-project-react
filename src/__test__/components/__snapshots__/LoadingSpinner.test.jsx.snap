// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoadingSpinner Component snapshot spinner 1`] = `
<div>
  <span
    class="relative w-12 h-12 rounded-full border-4 border-white border-t-transparent animate-spin inline-block box-border"
  >
    <span
      class="absolute inset-0 m-auto w-10 h-10 border-4 border-b-red-600 border-l-red-600 border-t-transparent border-r-transparent rounded-full animate-[spin_0.5s_linear_infinite]"
    />
    <span
      class="absolute inset-0 m-auto w-8 h-8 border-4 border-white border-t-transparent border-r-transparent rounded-full animate-[spin_1.5s_linear_infinite]"
    />
  </span>
</div>
`;
